src/test/librbd/mirror/snapshot/test_mock_UnlinkPeerRequest.cc

```diff
@@ -206,7 +236,34 @@ TEST_F(TestMockMirrorSnapshotUnlinkPeerRequest, SnapshotRemoveEmptyPeers) {
 
   C_SaferCond ctx;
   auto req = new MockUnlinkPeerRequest(&mock_image_ctx, snap_id, "peer_uuid",
-                                       &ctx);
+                                       true, &ctx);
+  req->send();
+  ASSERT_EQ(0, ctx.wait());
+}
+
+TEST_F(TestMockMirrorSnapshotUnlinkPeerRequest, SnapshotRemoveEmptyPeersNotAllowed) {
+  REQUIRE_FORMAT_V2();
+
+  librbd::ImageCtx *ictx;
+  ASSERT_EQ(0, open_image(m_image_name, &ictx));
+
+  MockTestImageCtx mock_image_ctx(*ictx);
+  cls::rbd::MirrorSnapshotNamespace ns{
+    cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY, {},
+    "", CEPH_NOSNAP};
+  auto snap_id = snap_create(mock_image_ctx, ns, "mirror_snap");
```

Comment:
`SnapshotRemoveEmptyPeers` creates a second snapshot with a peer (i.e. it's not peer-less like the first one).  I don't think it really matters in either case but let's repeat that here, for consistency:
```suggestion
  auto snap_id = snap_create(mock_image_ctx, ns, "mirror_snap");
  ns.mirror_peer_uuids = {"peer_uuid"};
```

Reply:
Sure. I removed it because it seemed out of place with the "EmptyPeers" in the title, but perhaps there's a difference I am not considering.